---
title: C6318 | Documents Microsoft
ms.custom: ''
ms.date: 11/04/2016
ms.technology:
- vs-ide-code-analysis
ms.topic: conceptual
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
author: mikeblome
ms.author: mblome
manager: douge
ms.workload:
- multiple
ms.openlocfilehash: deb8de9e766282b305ddce61157a32a3c2585b35
ms.sourcegitcommit: 6a9d5bd75e50947659fd6c837111a6a547884e2a
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 04/16/2018
---
# <a name="c6318"></a>C6318
avertissement C6318 : __try mal définie /\__except : utilisation de la constante EXCEPTION_CONTINUE_SEARCH ou d’une autre constante égale à zéro dans l’expression de filtre d’exception. Le code dans le bloc de gestionnaire d’exceptions n’est pas exécuté.  
  
 Cet avertissement indique que si une exception se produit dans le bloc protégé de ce gestionnaire d’exceptions structuré, l’exception n’est pas gérée car la constante `EXCECPTION_CONTINUE_SEARCH` est utilisé dans l’expression de filtre d’exception.  
  
 Ce code équivaut au bloc protégé sans le bloc de gestionnaire d’exceptions, car le bloc de gestionnaire n’est pas exécuté.  
  
## <a name="example"></a>Exemple  
 Le code suivant génère cet avertissement :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Pour corriger cet avertissement, utilisez le code suivant :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>Voir aussi  
 [try-except, instruction](/cpp/cpp/try-except-statement)